Concepts.md

```diff
@@ -0,0 +1,121 @@
+
+**_Data Mode_**
+
+Prometheus về cơ bản lưu trữ tất cả dữ liệu theo time series : Time series được định danh duy nhất với metric. 
+Metric là sự kết hợp của name metric và dimension của nó. Trong đó dimension được xác định với label và value của nó. Label bắt đầu bằng _ được dành riêng cho sử dụng nội bộ.
+
+*Metric names and labels* 
+
+-   Name metric: chỉ ra đặc tính chung của hệ thống cần đo . Ví dụ:  `http_requests_total`  => tổng số HTTP request nhận được.
+
+-    PromQL sẽ dựa trên các dimension để lọc và tổng hợp dữ liệu. Ví dụ: với metric  `http_requests_total`, chúng ta có thể có 2 label là  `method="POST"`,  `handler="/messages"`
+
+*Sample*
+Các samples hình thành lên time series thực tế ,mỗi sample bao gồm :
+- một  giá trị float64 
+- một dấu thời gian chính xác đến mili giây.
+
+*Notation*
+```
+<metric name>{<label name>=<label value>, ...}
+```
+Ex:
+```
+api_http_requests_total{method="POST", handler="/messages"}
+```
+
+
+**Metric type**
+
+Prometheus hỗ trợ 4 metric type là:
+
+1.  Counter : 
+ Metric tích lũy đại diện cho một bộ đếm chỉ có thể tăng dần hoặc được đặt lại về 0 khi khởi động lại. Ví dụ: bạn có thể sử dụng bộ đếm để thể hiện số lượng yêu cầu được phục vụ, nhiệm vụ đã hoàn thành hoặc lỗi.
+Không sử dụng bộ đếm với giá trị có thể giảm. Ví dụ: không sử dụng bộ đếm cho số lượng quy trình hiện đang chạy nên sử dụng Gauge.
+
+2.  Gauge :
+Metric đại diện cho một giá trị số có thể tùy ý lên xuống. thường được sử dụng cho các giá trị đo như nhiệt độ hoặc mức sử dụng bộ nhớ hiện tại, và "counter" có thể tăng và giảm, như số lượng yêu cầu đồng thời.
+
+3.  Histogram :
+Biểu đồ được sử dụng để quan sát. Biểu đồ bao gồm sự kết hợp của các số liệu sau:
+Bucket : là counter của các quan sát. Nó nên có giới hạn trên và giới hạn dưới.
+	- `<basename> _bucket {le = "< upper includesive bound >"}` : bộ đếm tích lũy cho các bucket quan sát .
+	- `<basename> _sum`: tổng cộng của tất cả các giá trị được quan sát.
+	- `<basename> _count` : số lượng các sự kiện đã được quan sát .( = `<basename>_bucket{le="+Inf"}`)
+
+	Sử dụng hàm histogram_quantile () để tính toán lượng tử từ biểu đồ hoặc thậm chí tổng hợp biểu đồ.
+
+A Histogram looks like:
+
+`request_latency_seconds_bucket{le="0.025"} 4.0
+request_latency_seconds_bucket{le="0.05"} 7.0
+request_latency_seconds_bucket{le="0.075"} 10.0
+request_latency_seconds_bucket{le="1.0"} 10.0
+request_latency_seconds_bucket{le="+Inf"} 11.0
+request_latency_seconds_count 11.0
+request_latency_seconds_sum 3.3` 
+
+
+4.  Summary : 
+ Một phần mở rộng của Histogram. Bên cạnh cũng cung cấp tổng và số lượng quan sát , nó tính toán các lượng tử có thể cấu hình qua sliding time window.
+Một Summary hiển thị nhiều chuỗi thời gian trong một lần quét:
+	- `<basename> {quantile = "<φ>"}`:  φ-quantiles (0 ≤ φ≤ 1) các sự kiện được quan sát.
+	- `<basename> _sum`: tổng cộng của tất cả các giá trị được quan sát.
+	- `<basename> _count`: số lượng các sự kiện đã được quan sát .
+
+Histogram vs Summary : .... 
```

Comment:
đọc kĩ rồi so sánh xem nhé

Reply:
Tạm thời em xem được đến đây :

"  Một số khác biệt giữa Summary và Historgram:
    -  Với Historgram lượng tử được tính trên máy chủ Prometheus, với Summary chúng được tính trên máy chủ ứng dụng . Do đó, dữ liệu Summary không thể được tổng hợp từ một số trường hợp ứng dụng.
    -  Historgram yêu cầu định nghĩa giá trị trước của bucket , vì vậy phù hợp với trường hợp sử dụng khi có ý tưởng tốt về sự tràn dải của các giá trị.
    - Summary là một lựa chọn tốt nếu cần tính toán các lượng tử chính xác, nhưng không thể chắc chắn phạm vi của các giá trị sẽ là bao nhiêu. "

Reply:
If you use a summary, you control the error in the dimension of φ. If you use a histogram, you control the error in the dimension of the observed value (via choosing the appropriate bucket layout). With a broad distribution, small changes in φ result in large deviations in the observed value. With a sharp distribution, a small interval of observed values covers a large interval of φ.

Đọc phần này em hơi hiểu vì chưa hình dung được , anh giải thích giúp em với ạ .